plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1278,4 +1292,57 @@ public void testDeleteWithVarcharPredicate()
     {
         throw new SkipException("This is implemented by testDeleteWithVarcharEqualityPredicate");
     }
+
+    @Test(dataProvider = "testInsertBatchSizeSessionProperty")
+    public void testInsertBatchSizeSessionProperty(Integer batchSize, Integer numberOfRows)
+    {
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "insert_batch_size", batchSize.toString())
+                .build();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_insert_batch_size",
+                "(a varchar(36), b bigint)")) {
+
+            String values = String.join(",", makeValuesForInsertBatchSizeSessionPropertyTest(numberOfRows));
+            assertUpdate(session, "INSERT INTO " + table.getName() + " (a, b) VALUES " + values, numberOfRows);
+            assertQuery("SELECT COUNT(*) FROM " + table.getName(), format("VALUES %d", numberOfRows));
+        }
+    }
+
+    private static List<String> makeValuesForInsertBatchSizeSessionPropertyTest(int numberOfRows) {
+        List<String> result = new ArrayList<>(numberOfRows);
+        for (int i = 0; i < numberOfRows; i++) {
+            result.add(String.format("('%s', %d)", UUID.randomUUID(), ThreadLocalRandom.current().nextLong()));
+        }
+        return result;
+    }
+
+    @DataProvider(name = "testInsertBatchSizeSessionProperty")
+    public static Object[][] batchSizeAndNumberOfRowsForInsertBatchSizePropertyTest()
+    {
+        return new Object[][] {
+                {100, 10},
```

Comment:
How much time does all values take to run?

Maybe we just need the following cases (@kokosing @wendigo should be enough to cover the important cases?)

```
{100, <100},
{100, 100},
{100, >100},

{1000, <1000}
{1000, 1000}
{1000, >1000}

{10k, <10k},
{10k, 10k},
{10k, >10k}
```

Reply:
After cleaning up list of test data provider, local mvn test on pluginc/trino-base-jdbc runs ~3% (3 sec) longer:

mvn test  557.03s user 45.09s system 698% cpu 1:26.14 total
vs.
mvn test  596.36s user 45.72s system 720% cpu 1:29.17 total